### PR TITLE
Expose value field in Tag (since 9.2)

### PR DESCRIPTION
**Pull request by tommydeboer**

### CODE REVIEW

**Comment by tommydeboer** [comment]
[EMX-Downloader expose value field in sys_md_Tag](https://trello.com/c/eWzLlHwt/406-emx-downloader-expose-value-field-in-sysmdtag)

File: src/main/java/org/molgenis/downloader/api/metadata/MolgenisVersion.java [hunk 1]
```diff
@@ -54,18 +55,6 @@ public boolean equalsOrSmallerThan(final MolgenisVersion that) {
     return this.equals(that) || this.smallerThan(that);
   }
 
-  public int getMajor() {
-    return major;
-  }
-
```
Comment (jelmerveen) [inline src/main/java/org/molgenis/downloader/api/metadata/MolgenisVersion.java:60]: why is this removed?

File: src/main/java/org/molgenis/downloader/api/metadata/Tag.java [hunk 2]
```diff
@@ -61,29 +72,30 @@ public void setLabel(String label) {
...
+    }
+    if (o == null || getClass() != o.getClass()) {
       return false;
-    if (relationLabel != null
-        ? !relationLabel.equals(tag.relationLabel)
-        : tag.relationLabel != null) return false;
-    return codeSystem != null ? codeSystem.equals(tag.codeSystem) : tag.codeSystem == null;
+    }
+    Tag tag = (Tag) o;
+    return Objects.equals(getLabel(), tag.getLabel())
```
Comment (jelmerveen) [inline src/main/java/org/molgenis/downloader/api/metadata/Tag.java:82]: looks clean :)!

File: src/main/java/org/molgenis/downloader/api/metadata/Tag.java [hunk 3]
```diff
@@ -61,29 +72,30 @@ public void setLabel(String label) {
...
   @Override
   public int hashCode() {
-    int result = label != null ? label.hashCode() : 0;
-    result = 31 * result + (objectIRI != null ? objectIRI.hashCode() : 0);
-    result = 31 * result + (relationIRI != null ? relationIRI.hashCode() : 0);
-    result = 31 * result + (relationLabel != null ? relationLabel.hashCode() : 0);
-    result = 31 * result + (codeSystem != null ? codeSystem.hashCode() : 0);
-    return result;
+    return Objects.hash(
+        getLabel(),
```
Comment (jelmerveen) [inline src/main/java/org/molgenis/downloader/api/metadata/Tag.java:93]: no more need of null checks?

File: src/main/java/org/molgenis/downloader/api/metadata/MolgenisVersion.java (hunk 1 above)
Comment (tommydeboer) [reply src/main/java/org/molgenis/downloader/api/metadata/MolgenisVersion.java:60]: Unused code

File: src/main/java/org/molgenis/downloader/api/metadata/Tag.java (hunk 2 above)
Comment (tommydeboer) [reply src/main/java/org/molgenis/downloader/api/metadata/Tag.java:82]: It's the new way IntelliJ auto-generates the `equals` and `hashCode` methods 👌

File: src/main/java/org/molgenis/downloader/api/metadata/Tag.java (hunk 3 above)
Comment (tommydeboer) [reply src/main/java/org/molgenis/downloader/api/metadata/Tag.java:93]: I think the `Objects.hash()` takes care of that

File: src/main/java/org/molgenis/downloader/api/metadata/MolgenisVersion.java (hunk 1 above)
Comment (jelmerveen) [reply src/main/java/org/molgenis/downloader/api/metadata/MolgenisVersion.java:60]: are the private fields still relevant then?

File: src/main/java/org/molgenis/downloader/api/metadata/Tag.java (hunk 3 above)
Comment (jelmerveen) [reply src/main/java/org/molgenis/downloader/api/metadata/Tag.java:93]: did you test?

File: src/main/java/org/molgenis/downloader/api/metadata/MolgenisVersion.java (hunk 1 above)
Comment (tommydeboer) [reply src/main/java/org/molgenis/downloader/api/metadata/MolgenisVersion.java:60]: Yes, they're used in the other methods

File: src/main/java/org/molgenis/downloader/api/metadata/Tag.java (hunk 3 above)
Comment (tommydeboer) [reply src/main/java/org/molgenis/downloader/api/metadata/Tag.java:93]: Well... no. I trust IntelliJ 🙏

Comment (tommydeboer) [reply src/main/java/org/molgenis/downloader/api/metadata/Tag.java:93]: It's handled by `Arrays.hashCode()`:

```
    public static int hashCode(Object a[]) {
        if (a == null)
            return 0;

        int result = 1;

        for (Object element : a)
            result = 31 * result + (element == null ? 0 : element.hashCode());

        return result;
    }

**Review by jelmerveen — APPROVED** [review]
